screens/CreateEventScreen.jsx

```diff
@@ -0,0 +1,414 @@
+import { createEvent } from '../api/firebaseMethods';
+import * as ImagePicker from 'expo-image-picker';
+import * as ImageManipulator from 'expo-image-manipulator';
+import DefaultImg from '../assets/images/defaultprofile.png'
+import * as firebase from 'firebase';
+import { editProfile, uploadPicture } from '../api/firebaseMethods';
+import { follow, unfollow } from '../api/firebaseMethods'
+import { useUser, useUserUpdate } from '../hooks/UserContext';
+
+export default function CreateEventScreen({ navigation }) {
+  const [wordCount, setWordCount] = useState(0)
+
+  const [showName, setShowName] = useState('');
+  const [description, setDescription] = useState('');
+  const [tags, setTags] = useState([])
+
+  const [ticketPrice, setTicketPrice] = useState('');
+  const [ticketLink, setTicketLink] = useState('');
+  const [intermission, setIntermission] = useState(false);
+  const toggleIntermission = () => setIntermission(previousState => !previousState);
+  const [ticketRequired, setTicketRequired] = useState(false);
+  const toggleTicketRequired = () => setTicketRequired(previousState => !previousState);
+
+  const [venueName, setVenueName] = useState('');
+  const [venueAddress, setVenueAddress] = useState('');
+  const [wheelchair, setWheelchair] = useState(false)
+  const toggleWheelchair = () => setWheelchair(previousState => !previousState);
+  const [hearingAssistance, setHearingAssistance] = useState(false);
+  const toggleHearingAssistance = () => setHearingAssistance(previousState => !previousState);
+
+  const [cast, setCast] = useState([{name:'', role:''}]);
+  var castId = 0;
+
+  const [users, setUsers] = useState([]);
+  const currentUserInfo = useUser();
+  const updateUserInfo = useUserUpdate();
+  const currentUserFollowing = currentUserInfo.following;
+  const currentUserID = currentUserInfo.id;
+
+  const emptyState = () => {
+    setShowName('');
+    setDescription('');
+    setVenueName('');
+    setVenueAddress('');
+  };
+
+  const handlePress = () => {
+    if (!showName) {
+      Alert.alert('Show name is required!');
+    } else if (!venueName) {
+      Alert.alert('Venue name is required!');
+    } else if (!venueAddress) {
+      Alert.alert('Venue Address is required.');
+    } else if (wordCount > 250) {
+      Alert.alert('Description must be less than 250 words.')
+    } else {
+      handleAdd();
+      createEvent(
+        showName,
+        description,
+        venueName,
+        venueAddress,
+      );
+      navigation.navigate('Explore');
+      emptyState();
+      castId = 0;
+    }
+  };
+
+  const onChange = (t, e) => {
+    (e === 'role') ? (cast.length === castId ? setCast([...cast, {...(cast[castId]), role:t}]) : setCast([...(cast.slice(0,-1)), {...(cast[cast.length-1]), role:t}])) :
+    (cast.length === castId ? setCast([...cast, {...(cast[castId]), name:t}]) : setCast([...(cast.slice(0,-1)), {...(cast[cast.length-1]), name:t}]));
+  };
```

Comment:
i think this should take in an index field. also, rather than have `t` be the value and `e` be the type, consider having a name parameter and a role parameter, where you pass in null (empty string may cause problems) if you don't want to update that specific field. then, to update, you could do something like this:

`onChange(index, name, role):`

`name = name === null ?  cast[index].name : name`
`role = role === null ? cast[index].role : role`
`splice { name, role } into array`

i just hesitate to use a variable in different ways depending on another variable.